cql-parser/src/main/scala/troy/cql/parser/CqlParser.scala

```diff
@@ -142,10 +142,12 @@ object CqlParser extends JavaTokenParsers with Helpers {
       "WHERE".i ~> rep1sep(relation, "AND".i) ^^ WhereClause.apply
     }
 
+    def limit = "LIMIT" ~> """([1-9]\d+)""".r ^^ { _.toInt }
```

Comment:
Regarding the regex, I guess `"""\d+""".r` is enough. Also you might define it near the end of the files to be reused as `postiveNumber`.